app/scripts/migrations/121.1.ts

```diff
@@ -0,0 +1,66 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+import { CHAIN_IDS } from '../../../shared/constants/network';
+
+export const version = 121.1;
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+/**
+ * Migrates MATIC ticker in Network Configuration to POL ticker as per the direction in https://polygon.technology/blog/save-the-date-matic-pol-migration-coming-september-4th-everything-you-need-to-know
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>): void {
+  const networkControllerState = state.NetworkController;
+  if (
+    hasProperty(state, 'NetworkController') &&
+    isObject(networkControllerState) &&
+    hasProperty(networkControllerState, 'networkConfigurations') &&
+    isObject(networkControllerState.networkConfigurations)
+  ) {
+    for (const networkConfiguration of Object.values(
+      networkControllerState.networkConfigurations,
+    )) {
+      if (
+        isObject(networkConfiguration) &&
+        networkConfiguration.chainId === CHAIN_IDS.POLYGON &&
+        networkConfiguration.ticker === 'MATIC'
+      ) {
+        networkConfiguration.ticker = 'POL';
+      }
+    }
+  }
+
+  if (
+    hasProperty(state, 'NetworkController') &&
+    isObject(networkControllerState) &&
+    hasProperty(networkControllerState, 'providerConfig') &&
+    isObject(networkControllerState.providerConfig) &&
+    hasProperty(networkControllerState.providerConfig, 'chainId') &&
+    networkControllerState.providerConfig.chainId === CHAIN_IDS.POLYGON &&
+    networkControllerState.providerConfig.ticker === 'MATIC'
+  ) {
+    networkControllerState.providerConfig.ticker = 'POL';
+  }
+  state = {
+    ...state,
+    NetworkController: networkControllerState,
+  };
```

Comment:
This line has no effect, and can be omitted

```suggestion
```

The state changes are applied already by this point. And even if they weren't, this would have no effect because we're creating a new object here (which then is not returned or accessed ever).

Reply:
Thanks. This change is in this commit here: https://github.com/MetaMask/metamask-extension/pull/26671/commits/d9093413eb39dd1e24f322e8898cf205e962c50d